libs/execution_context/ebpf_link.c

```diff
@@ -109,15 +111,12 @@ ebpf_link_attach_program(ebpf_link_t* link, ebpf_program_t* program)
     }
 
     link->program = program;
-    ebpf_object_acquire_reference((ebpf_object_t*)program);
+    ebpf_program_add_link_to_list(program, link);
```

Comment:
The fact that we use a linked list should be internal to the program object. Instead, this should be something like:
ebpf_program_attach_link / ebpf_program_detach_link. 

Also, to be consistent the ebpf_program_attach_link/detach_link should do the ref-count adjustment.

Reply:
`ebpf_object_acquire_reference((ebpf_object_t*)program)` should be moved inside the new function as well.

Reply:
1. Updated function names to ebpf_program_attach_link / ebpf_program_detach_link 
2. Moved ebpf_object_acquire_reference / release_ref inside the above functions.